tools/perf/tests/shell/stat+shadow_stat.sh

```diff
@@ -14,7 +14,7 @@ test_global_aggr()
 {
 	perf stat -a --no-big-num -e cycles,instructions sleep 1  2>&1 | \
 	grep -e cycles -e instructions | \
-	while read num evt hash ipc rest
+	while read -r num evt ipc rest
```

Comment:
With shellcheck -S warning, this is what we see:
# shellcheck -S warning stat+shadow_stat.sh

In stat+shadow_stat.sh line 36:
		res=`printf "%.2f" $(echo "scale=6; $num / $cyc" | bc -q)`
                                   ^-- SC2046: Quote this to prevent word splitting.


In stat+shadow_stat.sh line 48:
	while read cpu num evt hash ipc rest
                               ^--^ SC2034: hash appears unused. Verify use (or export if used externally).


In stat+shadow_stat.sh line 70:
		res=`printf "%.2f" $(echo "scale=6; $num / $cyc" | bc -q)`
                                   ^-- SC2046: Quote this to prevent word splitting.

For more information:
  https://www.shellcheck.net/wiki/SC2034 -- hash appears unused. Verify use (...
  https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitt...

The change for "-r" can be removed in this patch.